gdnative-core/src/init.rs

```diff
@@ -0,0 +1,65 @@
+use crate::core_types::GodotString;
+
+pub struct InitCallbackInfo {
+    in_editor: bool,
+    active_library_path: GodotString,
+    options: *mut crate::sys::godot_gdnative_init_options,
+}
+
+impl InitCallbackInfo {
+    /// Returns true if the library is loaded in the Godot Editor.
+    #[inline]
+    pub fn in_editor(&self) -> bool {
+        self.in_editor
+    }
+
+    /// Returns a path to the library relative to the project.
+    ///
+    /// Example: res://../../target/debug/libhello_world.dylib
+    #[inline]
+    pub fn active_library_path(&self) -> &GodotString {
+        &self.active_library_path
+    }
+
+    /// # Safety
+    ///
+    /// Will `panic!()` if options is NULL or invalid.
+    #[doc(hidden)]
+    #[inline]
+    pub unsafe fn new(options: *mut crate::sys::godot_gdnative_init_options) -> Self {
+        assert!(!options.is_null(), "InitCallbackOptions were NULL");
+        let crate::sys::godot_gdnative_init_options {
+            in_editor,
+            active_library_path,
+            ..
+        } = *options;
+
+        let active_library_path =
+            crate::core_types::GodotString::clone_from_sys(*active_library_path);
+
+        Self {
+            options,
+            in_editor,
+            active_library_path,
+        }
+    }
+
+    #[inline]
+    pub fn report_loading_error(&self, message: &str) {
```

Comment:
If we convert to `CString` anyway, wouldn't it make sense to accept any `Display` type?